### ISSUE TITLE
Code for this tutorial didn't link

### ISSUE THREAD

**Issue by erz42** [MicrosoftDocs/cpp-docs#3630]
I am very interested in using modules and am happy Visual Studio is being so proactive about supporting them.  I tried to build the sample described in this page.  It doesn't link.  The first time I noticed the error I was using VS 17.0.0.  After seeing the error I upgraded to VS 17.0.4.  However, the build still doesn't work for me.  Maybe I missed a step?

Build started...
1>------ Build started: Project: ModulesTutorial, Configuration: Debug x64 ------
1>Scanning sources for module dependencies...
1>BasicPlane.Figures-Point.ixx
1>BasicPlane.Figures.ixx
1>BasicPlane.Figures-Rectangle.ixx
1>Compiling...
1>BasicPlane.Figures-Point.ixx
1>BasicPlane.Figures-Rectangle.ixx
1>BasicPlane.Figures.ixx
1>BasicPlane.Figures-Rectangle.cpp
1>ModulesTutorial.cpp
1>Generating Code...
1>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(1087,5): error MSB6006: "link.exe" exited with code 1120.
1>ModulesTutorial.obj : error LNK2019: unresolved external symbol "int __cdecl area(struct Rectangle const &)" (?area@@YAHAEBURectangle@@@Z::<!BasicPlane.Figures>) referenced in function main
1>  Hint on symbols that are defined and could potentially match:
1>    "int __cdecl area(struct Rectangle const &)" (?area@@YAHAEBURectangle@@@Z)
1>C:\Users\eric\source\repos\ModulesTutorial\x64\Debug\ModulesTutorial.exe : fatal error LNK1120: 1 unresolved externals
1>Done building project "ModulesTutorial.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 01648994-5c73-a88d-bc78-dbab40d834d7
* Version Independent ID: 01648994-5c73-a88d-bc78-dbab40d834d7
* Content: [Named modules tutorial in C++](https://docs.microsoft.com/en-us/cpp/cpp/tutorial-named-modules-cpp?view=msvc-170)
* Content Source: [docs/cpp/tutorial-named-modules-cpp.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/cpp/tutorial-named-modules-cpp.md)
* Product: **visual-cpp**
* Technology: **cpp-language**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by erz42** [comment]
[ModulesTutorial.zip](https://github.com/MicrosoftDocs/cpp-docs/files/7799868/ModulesTutorial.zip)

**Comment by corob-msft** [comment]
#3615 duplicate

**Comment by corob-msft** [comment]
@erz42 

Thanks for the report. The code didn't link for me in 17.0.4 either, but then I noticed this line in the tutorial:

> For this tutorial, you'll need Visual Studio 2022 17.1.0 or later.

I verified that the tutorial does in fact work in the 17.1 preview builds. We don't usually publish tutorials until the compiler they require is released, so someone may have poked the publish button on this article too soon. Possibly me. Apologies for the misuse of your time.

I'm closing the issue as 17.1 previews are available now and the 17.1 release is coming shortly, probably within a couple of weeks.
